Investigations/MIMIC_Notebook.ipynb

```diff
@@ -0,0 +1,531 @@
+    "\n",
+    "Note that not all of these will work, some are hit and miss. We predominantly rely on continuous and discrete KL divergence measures. You can also input <b>\"gower\"</b> and this will calculate the gower distance using the gower library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d845b202",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- Datetime Handling -------- #\n",
+    "\n",
+    "# If the dataset has datetimes then we need to re-convert these to a numerical\n",
+    "# Value representing seconds, this is so we can evaluate the metrics on them\n",
+    "\n",
+    "metric_synthetic_supp = synthetic_supp.copy()\n",
+    "\n",
+    "for index, column in enumerate(original_datetime_columns):\n",
+    "\n",
+    "    # Fit datetime transformer - converts to seconds\n",
+    "    temp_datetime = datetime.DatetimeTransformer()\n",
+    "    temp_datetime.fit(metric_synthetic_supp, columns=column)\n",
+    "\n",
+    "    metric_synthetic_supp = temp_datetime.transform(metric_synthetic_supp)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f7e4174b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- SDV Metrics -------- #\n",
+    "\n",
+    "# Define the metrics you want the model to evaluate\n",
+    "\n",
+    "gower = False\n",
+    "\n",
+    "metrics = distribution_metrics(\n",
+    "    gower_bool=gower,\n",
+    "    data_supp=original_metric_set,\n",
+    "    synthetic_supp=metric_synthetic_supp,\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    saving_filepath=\"\",\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "975c5d80",
+   "metadata": {},
+   "source": [
+    "# Privacy Metric Evaluation\n",
+    "\n",
+    "Using SDV privacy metrics we can get an insight into how privacy is conserved when utilising dp-sgd methods. SDV's privacy metrics are limited in that they can only be used on similar data types. E.g. if we choose age to be the sensitive variable, we can build ML based models to predict a users age using the other columns. However we are forced to only use columns that are also continuous variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3e4c9495",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Specify our private variable\n",
+    "\n",
+    "private_variable = \"x14\"\n",
+    "\n",
+    "privacy_metric = privacy_metrics(\n",
+    "    private_variable=private_variable,\n",
+    "    data_supp=data_supp,\n",
+    "    synthetic_supp=synthetic_supp,\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    saving_filepath=None,\n",
+    "    pre_proc_method=pre_proc_method,\n",
```

Comment:
Possibly add a print of the metrics here to show the outputs in the notebook as well